packages/shared/src/components/tooltips/ChangelogTooltip.tsx

```diff
@@ -0,0 +1,167 @@
+                loading="lazy"
+                data-testid="changelogImage"
+              />
+              <h3
+                className="mt-2 font-bold text-theme-label-primary typo-title3"
+                data-testid="changelogTitle"
+              >
+                {post.title}
+              </h3>
+              <time
+                className="text-theme-label-quaternary typo-callout"
+                dateTime={post.createdAt}
+                data-testid="changelogDate"
+              >
+                {postDateFormat(post.createdAt)}
+              </time>
+              <div className="mt-2 w text-theme-label-tertiary typo-callout">
+                {post.summary}
+              </div>
+            </section>
+            <footer className="flex gap-3 items-center py-3 px-4 w-full h-16 border-t border-theme-divider-tertiary">
+              <Button
+                className="btn-tertiary"
+                onClick={dismissChangelog}
+                tag="a"
+                href={post.permalink}
+                data-testid="changelogReleaseNotesBtn"
+              >
+                Release notes
+              </Button>
+              {isExtension && (
+                <Button
+                  tag={isFirefoxExtension ? 'a' : undefined}
+                  href={
+                    isFirefoxExtension
+                      ? 'https://r.daily.dev/firefoxupdate'
+                      : undefined
+                  }
+                  className="bg-cabbage-40 btn-primary"
+                  data-testid="changelogExtensionBtn"
+                  onClick={async () => {
+                    if (isFirefoxExtension) {
+                      return;
+                    }
+
+                    if (isExtension) {
+                      const sendMessage =
+                        globalThis?.browser?.runtime?.sendMessage;
+
+                      if (typeof sendMessage === 'function') {
+                        const updateResponse: { status: string } =
+                          await sendMessage({
+                            type: ExtensionMessageType.RequestUpdate,
+                          });
+
+                        let toastMessage: string;
+
+                        switch (updateResponse.status) {
+                          case 'throttled':
+                            toastMessage =
+                              'There is no update available, try again later';
+                            break;
+                          case 'no_update':
+                            toastMessage =
+                              'You are already on the latest available version';
+                            break;
+                          case 'update_available':
+                            toastMessage = 'Browser extension updated';
+                            break;
+                          default:
+                            break;
+                        }
```

Comment:
I think we can create an object with key as the `status` and the value will be what we set for the toast message. It would also probably a good idea to turn it into enum since we will be creating a map for the message to be type-strict. WDYT?

Then we can make it like below or something similar:
```
const toastMessage = updateStatusMessage[updateResponse.status]
```

Reply:
I though about that but in the end I don't have the correct types because of the outdated `webextension-polyfill` and I think it would be contra productive to add our own `status` type. When #733 is merged I think it would be a right time to then just use https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/RequestUpdateCheckStatus 

I did however refactor to object map, it is more readable like this 👍